snet_cli/mpe_service_metadata.py

```diff
@@ -0,0 +1,101 @@
+# functions for manipulate with server.json file
+# metadata format:
+#----------------------------------------------------
+# version         - used to track format changes (current version is 1)
+# model_ipfs_hash - IPFS HASH to the .tar archive of protobuf service specification
+# mpe_address     - Address of MultiPartyEscrow contract. 
+#                   Client should use it exclusively for cross-checking of mpe_address, 
+#                        (because service can attack via mpe_address)
+#                   Daemon can use it directly if authenticity of metadata is confirmed
```

Comment:
Side note on keeping MultiPartyEscrow (MPE) contract address in metadata.

MPE contract address is also required to correctly handle MPE contracts updates in both client and service sides. MPE contract address is used in signed payment message to guarantee that both payment sender and payment receiver use the same data source to check the payment.

When data is kept in contract then contract address changes during contract upgrade and service provider should publish new Service with new metadata to incorporate change.

When data and logic of contract are decoupled the upgrade may be seamless for sender and receiver as most of upgrades doesn't affect data contract address. And changing this address in metadata is required only when data part of the contract is updated.

Reply:
we should discuss it here https://github.com/singnet/wiki/blob/master/multiPartyEscrowContract/MPEServiceMetadata.md